added getTables() for a more flexible public api

selfup opened the pull request:
Can return a where equivalent of tables.

This way the public API has a bit more flexibility for developers if they need two or four tables at the same time.

selfup committed:
tested double and quad get for a more flexible public api

selfup committed:
Added gitignore steps in README - changed formatting

selfup committed:
updated short description

selfup committed:
better README format

selfup committed:
testing out .gitignore

JoshCheek commented:
How about generic `getTables(...tables)` which allows you to pass as many as you like.

selfup commented:
@JoshCheek That is a much better idea! Thanks!

selfup committed:
refactored to taking in an array:

Now the number of tables is not limited.
The dev will need to pass an array of tableNames in strings.

selfup commented:
@JoshCheek 

The `...tables` syntax does not seem to work in ES6 classes?

I tried everything.

Now you will just have to pass it an Array, which could be easier for developers if they automate/build an array based on all tables created or something!

Let me know what you think

selfup committed:
added deper description in package.json

JoshCheek commented:
It's es-something syntax. I think the equivalent in es5 is:

``` javascript
> (function() { return Array.prototype.slice.call(arguments) })(1, 2, 3)
[ 1, 2, 3 ]
```

selfup committed:
Array.from(arguments) seems to have done the trick

selfup commented:
@JoshCheek 

```
getTables() {
    let tables = []
    for (let val of Array.from(arguments)) {
      tables.push(this.getTable(val))
    }
    return tables
  }
```

`Array.from(arguments)` seems to have done the trick!

selfup commented:
The test now passes without giving the method an array!

`this.rejs.getTables('firstTable', 'secondTable', 'thirdTable', 'fourthTable')`

selfup committed:
can do even and odd numbers - tests prove felxibility

selfup committed:
more readable 'for of' statement

selfup committed:
Added test coverage to README as well as how to run the tests with coverage

selfup commented:
Figured out how to get test coverage.

100% according to istanbul

@JoshCheek I think this might be ready!

JoshCheek commented:
At this point, you can probably just use map.

``` javascript
> (function() { return Array.from(arguments).map(n => 2*n) })(1, 2, 3)
[ 2, 4, 6 ]
```

If you want to keep the for loop, then you might not need to turn it into an array, IDK if that's better, but uses a for loop anyway:

``` javascript
> var arguments = { '0': 1, '1': 2, '2': 3 };
> var result = [];
> for(var i=0; i < arguments.length; ++i) {
    result.push(arguments[i]*2);
  }
> result
[ 4, 8, 12 ]
```

selfup commented:
for loops are technically faster, but I can try out the map!

selfup commented:
```
  getTables() {
    let tables = []
    Array.from(arguments).map(table => tables.push(this.getTable(table)))
    return tables
  }
```

This seems to work. However doing:

`return Array.from(arguments).map(table => tables.push(this.getTable(table)))`

Just returns: 1 2 3 4

selfup commented:
Wait I think I see what you did. Give me second!

selfup commented:
```
getTables() {
  return Array.from(arguments).map(table => this.getTable(table))
}
```

Works perfect!

selfup committed:
refactored getTables - thanks JoshCheek

selfup committed:
added lines hit coverage from istanbul

selfup committed:
Formatting in README

selfup commented:
@JoshCheek @adamki 

100% coverage and refactored.

If you wanna merge it, go ahead!

selfup committed:
added folder structure back so that the README makes sense

selfup committed:
reverted gitignore - Updated README to make sense about the rm -rf selfup-rejs

selfup committed:
better formatting for the README

selfup commented:
Thanks!
